### PR TITLE
Jane 6 upgrade

### PR DESCRIPTION
**Pull request by xavierlacot**
Upgrade to [JanePHP 6](https://github.com/janephp/janephp), which has been published 3 months ago. Also, added tests for the `JsonSorter` class.

### CODE REVIEW

File: generated/Model/FilesInfoGetResponse200.php [hunk 1]
```diff
@@ -16,11 +16,11 @@
 class FilesInfoGetResponse200
 {
     /**
-     * @var mixed[]|null
+     * @var |null
```
Comment (damienalexandre) [inline generated/Model/FilesInfoGetResponse200.php:19]: Is that a bug in Jane? It looks wrong.

**Comment by damienalexandre** [comment]
See https://jane.readthedocs.io/en/latest/components/OpenAPI.html#authentication and https://github.com/jolicode/slack-php-api/blob/master/src/ClientFactory.php#L42 :)

**Comment by xavierlacot** [comment]
> See https://jane.readthedocs.io/en/latest/components/OpenAPI.html#authentication and https://github.com/jolicode/slack-php-api/blob/master/src/ClientFactory.php#L42 :)

The `oauth2` security scheme type is not supported in Jane 6, it "only" supports `apiKey` and `http` (basic|bearer).

The only thing that we can do on this for now is patching the `securityDefinitions.slackAuth` spec node, in order to change its type:

```json
{
  "type": "apiKey",
  "in": "header",
  "name": "Authorization"
}
```

That way, Jane would generate an Authentication class, which basically only adds the header to all requests, which is exactly what we are doing with `HeaderAppendPlugin`. That said, I do not see this as significant enough improvement to warrant another patch to the specification...

**Comment by tecbird** [comment]
+1 please merge, i use another other clients which uses already jane 6.x

**Comment by xavierlacot** [comment]
Any news on this topic? Are there things that need to be sorted out before a merge?

**Comment by damienalexandre** [comment]
Yes my comment about the bad phpdoc is still relevant.

> generated/Model/FilesInfoGetResponse200.php

```diff
     /**
-     * @var mixed[]|null
+     * @var |null
```

I'm working with Jane Core Team to fix that up. In the meantime @tecbird you can require the client like this in your composer.json file:

```json
"jolicode/slack-php-api": "dev-jane-6-upgrade"
```

It would be great if you can give us feedback too! 

Cheers

**Comment by xavierlacot** [comment]
@damienalexandre the weird phpdoc has been fixed (thanks @Korbeil) and the tests are green :tada:

**Comment by damienalexandre** [comment]
Thanks a lot for your hard work both @Korbeil and @xavierlacot :)

**Comment by damienalexandre** [comment]
Version 3.0 is released, thanks!

cc @tecbird

**Pull request merged**